credentials/fetcher_default_test.go

```diff
@@ -153,6 +153,14 @@ func TestFetcherDefault(t *testing.T) {
 		assert.True(t, check("8e884167-1300-4f58-8cc1-81af68f878a8"))
 	})
 
+	time.Sleep(maxWait * 7) // wait so the fetched key reaches ttl
+
+	t.Run("name=should find the previously fetched key if the refresh request times out", func(t *testing.T) {
+		key, err := s.ResolveKey(context.Background(), uris, "c61308cc-faef-4b98-99c3-839f513ac296", "sig")
```

Comment:
How do we know if the key returned here is actually from the cache and not from the HTTP request which fetched it again?

Reply:
Nice catch, the test was a little bit shady - I improved it.